src/multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -979,11 +1066,145 @@ private void ClearSelectedAction()
+                energyBalance);
+        }
+
+        energyBalanceInfo = energyBalance;
+
+        organismStatisticsPanel.UpdateEnergyBalance(energyBalance);
+
+        float nominalStorage = 0;
+        Dictionary<Compound, float>? specificStorages = null;
+
+        // This takes balanceType into account as well, https://github.com/Revolutionary-Games/Thrive/issues/2068
+        var compoundBalanceData =
+            CalculateCompoundBalanceWithMethod(organismStatisticsPanel.BalanceDisplayType,
+                organismStatisticsPanel.CompoundAmountType,
+                cells, conditionsData, energyBalance,
+                ref specificStorages, ref nominalStorage);
+
+        UpdateCompoundBalances(compoundBalanceData);
+
+        // TODO: should this skip on being affected by the resource limited?
+        var nightBalanceData = CalculateCompoundBalanceWithMethod(organismStatisticsPanel.BalanceDisplayType,
+            CompoundAmountType.Minimum, cells, conditionsData, energyBalance, ref specificStorages,
+            ref nominalStorage);
+
+        UpdateCompoundLastingTimes(compoundBalanceData, nightBalanceData, nominalStorage,
+            specificStorages ?? throw new Exception("Special storages should have been calculated"));
+
+        HandleProcessList(cells, energyBalance, conditionsData);
+    }
+
+    private Dictionary<Compound, CompoundBalance> CalculateCompoundBalanceWithMethod(BalanceDisplayType calculationType,
+        CompoundAmountType amountType,
+        IReadOnlyList<HexWithData<CellTemplate>> cells, IBiomeConditions biome, EnergyBalanceInfo energyBalance,
+        ref Dictionary<Compound, float>? specificStorages, ref float nominalStorage)
+    {
+        Dictionary<Compound, CompoundBalance> compoundBalanceData = new();
+        foreach (var cell in cells)
+        {
+            switch (calculationType)
+            {
+                case BalanceDisplayType.MaxSpeed:
+                    ProcessSystem.ComputeCompoundBalance(cell.Data!.Organelles, biome,
+                        amountType, true, compoundBalanceData);
+                    break;
+                case BalanceDisplayType.EnergyEquilibrium:
+                    ProcessSystem.ComputeCompoundBalanceAtEquilibrium(cell.Data!.Organelles, biome,
+                        amountType, energyBalance, compoundBalanceData);
+                    break;
+                default:
+                    GD.PrintErr("Unknown compound balance type: ", organismStatisticsPanel.BalanceDisplayType);
+                    goto case BalanceDisplayType.EnergyEquilibrium;
+            }
+        }
+
+        specificStorages ??= CellBodyPlanInternalCalculations.GetTotalSpecificCapacity(cells.Select(o => o.Data!),
+            out nominalStorage);
+
+        return ProcessSystem.ComputeCompoundFillTimes(compoundBalanceData, nominalStorage, specificStorages);
+    }
+
+    private IEnumerable<(CellType Type, int Count)> GetCellTypes()
+    {
+        cellTypesWorkMemory.Clear();
```

Comment:
This seems a bit weird to me... this method always allocates an enumerator, so I think it would make a bit more sense to just rework the temporary storage to just act as the result. Something like `Dictionary<..., int>` should work, I think.